ocp_utilities/infra.py

```diff
@@ -354,3 +357,64 @@ def create_icsp(icsp_name, repository_digest_mirrors):
     )
     icsp.deploy()
     return icsp
+
+
+def dict_base64_encode(_dict):
+    """
+    Encoding dict in base64
+
+    Args:
+        _dict (dict): data dict to be encoded
+
+    Returns:
+        str: given _dict encoded in base64
+    """
+    return base64.b64encode(json.dumps(_dict).encode("ascii")).decode("utf-8")
+
+
+def create_update_secret(secret_data_dict, name, namespace):
+    """
+    Update existing secret or create a new secret; secret type - dockerconfigjson
+
+    Args:
+        secret_data_dict (dict): Secret data to be added/created,
+            example: {"auths":{
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>},
+                    ...,
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>}
+                    }
+                }
+        name (str): Secret name
+        namespace (str): Secret namespace
+
+    Returns:
+        Secret: secret object with secret_data_dict content included
+    """
+    secret = Secret(name=name, namespace=namespace)
+    secret_key = ".dockerconfigjson"
+    auths_key = "auths"
+
+    if secret.exists:
+        old_secret_data_dict = json.loads(
+            base64.b64decode(secret.instance.data[secret_key])
+        )[auths_key]
+        old_secret_data_dict.update(secret_data_dict[auths_key])
+        secret_data_encoded = dict_base64_encode(
+            _dict={auths_key: old_secret_data_dict}
+        )
+
+        ResourceEditor(
+            patches={secret: {"data": {secret_key: secret_data_encoded}}}
+        ).update()
+
+        return secret
+
+    secret_data_encoded = dict_base64_encode(_dict=secret_data_dict)
+    secret.res.update({"data": {secret_key: secret_data_encoded}})
```

Comment:
secret.data_dict = ...

Reply:
Getting an error when trying to modify `secret.data_dict` directly like this:
```
secret.data_dict = {"data": {secret_key: secret_data_encoded}}
```

The http response:
```
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Secret in version \\"v1\\" cannot be handled as a Secret: json: cannot unmarshal object into Go struct field Secret.data of type []uint8","reason":"BadRequest","code":400}

Reply:
check the structure of what you are trying to create; you have an unnecessary level

Reply:
right (:
missed that.
done